website/docs/Use-Cases/agent_chat.md

```diff
@@ -77,6 +77,62 @@ By adopting the conversation-driven control with both programming language and n
 - LLM-based function call. In this approach, LLM decides whether or not to call a particular function depending on the conversation status in each inference call.
   By messaging additional agents in the called functions, the LLM can drive dynamic multi-agent conversation. A working system showcasing this type of dynamic conversation can be found in the [multi-user math problem solving scenario](https://github.com/microsoft/autogen/blob/main/notebook/agentchat_two_users.ipynb), where a student assistant would automatically resort to an expert using function calls.
 
+  We register functions to enable function calls using the following two function decorators:
+
+  1. [`ConversableAgent.register_for_llm`](../reference/agentchat/conversable_agent#register_for_llm) is used to register the function in the `llm_config` of a ConversableAgent. The ConversableAgent agent can propose execution of a registrated function, but the actual execution will be performed by a UserProxy agent.
+
+  2. [`ConversableAgent.register_for_execution`](../reference/agentchat/conversable_agent#register_for_execution) is used to register the function in the `function_map` of a UserProxy agent.
+
+  The following examples illustrates the process of registering a custom function for currency exchange calculation:
+
+  ``` python
+  from typing_extensions import Annotated
+  from somewhere import exchange_rate
+
+  @user_proxy.register_for_execution()
+  @agent.register_for_llm(description="Currency exchange calculator.")
+  def currency_calculator(
+    base_amount: Annotated[float, "Amount of currency in base_currency"],
+    base_currency: Annotated[Literal["USD", "EUR"], "Base currency"] = "USD",
+    quote_currency: Annotated[Literal["USD", "EUR"], "Quote currency"] = "EUR",
+  ) -> str:
+    quote_amount = exchange_rate(base_currency, quote_currency) * base_amount
+    return f"{quote_amount} {quote_currency}"
```

Comment:
I think there should be more comment here (a line by line explanation) to explain how one would define a function. 

```
 @user_proxy.register_for_execution() # decorator that would register a function that can be executed by user_proxy agent. Specifically, adding {"currency_calculator": currency_calculator} to `function_map`.
  
base_amount: Annotated[float, "Amount of currency in base_currency"], # Annotated[<type of the param>, <description of the param>]
```